lib/archethic/transaction_chain/transaction/data.ex

```diff
@@ -91,6 +116,17 @@ defmodule Archethic.TransactionChain.TransactionData do
     {recipients, rest} =
       reduce_recipients(rest, nb_recipients, [], tx_version, serialization_mode)
 
+    # no need to check for serialization_mode because we never deserialize(:extended)
+    code =
+      try do
+        decompress_code(code)
```

Comment:
I noticed you use gzip to compress file so you could check first bytes instead of a try rescue